### ISSUE TITLE
Documentation has incorrect rendering of spaces for the indent-string option

### ISSUE THREAD

**Issue by BenjaminCarpenter480** [pylint-dev/pylint#8392]
### Bug description

In the documentation for the indent string option the code 
```
# String used as indentation unit. This is usually "  " (4 spaces) or "\t" (1
# tab).
indent-string="    "  
```
Is rendered incorrectly due to markdown or html folding multiple whitespaces into one i.e. (the "    " into " "). As such the documentation currently appears as:
![image](https://user-images.githubusercontent.com/6034496/223264469-6d09cf77-3aac-4a0d-ab57-78dc980944f8.png)

This can create a bit of confusion when attempting to change this option, as seen in https://stackoverflow.com/q/54117674/7052741

### Command used

```shell
N/A
```

### Pylint output

```shell
N/A
```

### Expected behavior

Something along the lines of:

### --indent-string
_String used as indentation unit. This is usually "&nbsp;&nbsp;&nbsp;&nbsp; " (4 spaces) or "\t" (1 tab)._
**Default**: "&nbsp;&nbsp;&nbsp;&nbsp;" 

### Pylint version

```shell
Appears on the newest version of Pylint taken from GitHub
```

**Comment by Pierre-Sassoulas** [comment]
Thank you for opening the issue. A fix would ideally fix all the value containing spaces in any option by using the proper rst rendering during generation.

**Comment by khushi-parikh** [comment]
Can I take up this issue? Currently the only fix I can find is using a code block for the part as whitespaces are preserved within it.

**Comment by Pierre-Sassoulas** [comment]
Sure, I'm assigning this issue to you :)

**Comment by locaaustin** [comment]
Hi. I saw this has been open for awhile and I would like to give it a go on fixing this issue if @khushi-parikh no longer wants to work on it. Thnx!

**Comment by wjandrea** [comment]
Thanks for raising this!

I'm not sure where you got the details though. I'm seeing 4 spaces in the RST, but you're showing 2 and it's not RST. BTW I'm seeing 1 space on the webpage, like in your screenshot.

### Here's what I see

#### [RST](https://pylint.readthedocs.io/en/stable/_sources/user_guide/configuration/all-options.rst.txt)

```rst
--indent-string
"""""""""""""""
*String used as indentation unit. This is usually "    " (4 spaces) or "\t" (1 tab).*

**Default:**  ``    ``
```

#### [Webpage](https://pylint.readthedocs.io/en/stable/user_guide/configuration/all-options.html#indent-string) - For reference

> ### --indent-string
> String used as indentation unit. This is usually " " (4 spaces) or "t" (1 tab).
> **Default:** \`\` \`\`

(Problems: 1 space, missing backslash before "t", and raw backticks for default)

HTML

```html

<h3>--indent-string<a class="headerlink" href="#indent-string" title="Link to this heading">¶</a></h3>
<p><em>String used as indentation unit. This is usually "    " (4 spaces) or "t" (1 tab).</em></p>
<p><strong>Default:</strong>  ``    ``</p>
```

### Output

As for the output, it should be monospaced, maybe like this:

> ### --indent-string
> String used as indentation unit. This is usually `"    "` (4 spaces) or `"\t"` (1 tab).
> **Default:** `"    "`

**Comment by Pierre-Sassoulas** [comment]
Please don't use an LLM for this, you should be a human and understand what you're doing. This is the kind of issue that we could fix ourselves pretty fast but we're keeping it for would-be junior contributors. Doing it with LLM and making us review the slop have negative value.

**Comment by locaaustin** [comment]
Yes, I am still planning to get started on this soon and won’t use LLMs.

**Comment by wjandrea** [comment]
@Pierre-Sassoulas Sorry, are you talking to me? I didn't use an LLM and I'm not really sure why you think I did.

**Comment by wjandrea** [comment]
OK, after reading back my message, I guess I can see why you would think that. I was really meticulous and used a peppy tone, but that's because OP seems to have gotten some details wrong and messed up some formatting, but I didn't want to rip them to shreds. Plus I don't really understand how RST works, so I wanted to explain whatever I could.
